Modules/gcmodule.c

```diff
@@ -64,74 +121,58 @@ _PyGC_Initialize(struct _gc_runtime_state *state)
 #define _GEN_HEAD(n) (&state->generations[n].head)
     struct gc_generation generations[NUM_GENERATIONS] = {
         /* PyGC_Head,                                 threshold,      count */
-        {{{_GEN_HEAD(0), _GEN_HEAD(0), 0}},           700,            0},
-        {{{_GEN_HEAD(1), _GEN_HEAD(1), 0}},           10,             0},
-        {{{_GEN_HEAD(2), _GEN_HEAD(2), 0}},           10,             0},
+        {{{_GEN_HEAD(0), (uintptr_t)_GEN_HEAD(0)}},   700,            0},
+        {{{_GEN_HEAD(1), (uintptr_t)_GEN_HEAD(1)}},   10,             0},
+        {{{_GEN_HEAD(2), (uintptr_t)_GEN_HEAD(2)}},   10,             0},
     };
     for (int i = 0; i < NUM_GENERATIONS; i++) {
         state->generations[i] = generations[i];
     };
     state->generation0 = GEN_HEAD(0);
     struct gc_generation permanent_generation = {
-          {{&state->permanent_generation.head, &state->permanent_generation.head, 0}}, 0, 0
+          {{&state->permanent_generation.head, (uintptr_t)&state->permanent_generation.head}}, 0, 0
     };
     state->permanent_generation = permanent_generation;
 }
 
 /*--------------------------------------------------------------------------
-gc_refs values.
+gc_prev values.
 
-Between collections, every gc'ed object has one of two gc_refs values:
+Between collections, gc_prev is used for doubly linked list.
 
-GC_UNTRACKED
-    The initial state; objects returned by PyObject_GC_Malloc are in this
-    state.  The object doesn't live in any generation list, and its
-    tp_traverse slot must not be called.
+Lowest three bits of gc_prev are used for flags.
+MASK_COLLECTING and MASK_TENTATIVELY_UNREACHABLE are used only while collecting.
```

Comment:
Can you explain that `MASK_COLLECTING` marks objects in the generation being collected?

Reply:
done